node/forest_libp2p/src/rpc/blocksync_message.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use forest_blocks::BlockHeader;
+use forest_cid::Cid;
+use forest_encoding::{
+    de::{self, Deserialize, Deserializer},
+    ser::{self, Serialize, Serializer},
+};
+use forest_message::{SignedMessage, UnsignedMessage};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Message {
+    pub start: Vec<Cid>,
+    pub request_len: u64,
+    pub options: u64,
+}
+
+impl Serialize for Message {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let value = (self.start.clone(), self.request_len, self.options);
+        ser::Serialize::serialize(&value, serializer)
+    }
+}
+impl<'de> Deserialize<'de> for Message {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let (start, request_len, options) = Deserialize::deserialize(deserializer)?;
+        Ok(Message {
+            start,
+            request_len,
+            options,
+        })
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Response {
+    pub chain: Vec<TipSetBundle>,
+    pub status: u64,
+    pub message: String,
+}
+
+impl Serialize for Response {
+    fn serialize<S>(&self, serializer: S) -> Result<<S as Serializer>::Ok, <S as Serializer>::Error>
+    where
+        S: Serializer,
+    {
+        let value = (self.chain.clone(), self.status, self.message.clone());
+        Serialize::serialize(&value, serializer)
+    }
+}
+impl<'de> Deserialize<'de> for Response {
+    fn deserialize<D>(deserializer: D) -> Result<Self, <D as Deserializer<'de>>::Error>
```

Comment:
```suggestion
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
```

Can you clean these up as well (on all of the serialize and deserialize fns for consistency)?

Reply:
fixed